aws/timeout_read_closer_test.go

```diff
@@ -53,3 +57,42 @@ func TestTimeoutReadCloserSameDuration(t *testing.T) {
 		t.Log(err)
 	}
 }
+
+func TestResponseReadTimeoutMiddleware(t *testing.T) {
+	reader := testReader{
+		duration: time.Second,
+		count:    0,
+	}
+
+	m := &readTimeoutMiddleware{duration: time.Millisecond}
+
+	out, _, err := m.HandleDeserialize(context.Background(),
+		middleware.DeserializeInput{},
+		middleware.DeserializeHandlerFunc(
+			func(ctx context.Context, input middleware.DeserializeInput) (
+				out middleware.DeserializeOutput, metadata middleware.Metadata, err error,
+			) {
+				response := &smithyhttp.Response{
+					Response: &http.Response{Body: &reader},
+				}
+				return middleware.DeserializeOutput{RawResponse: response}, metadata, err
+			}),
+	)
+
+	if err != nil {
+		t.Fatalf("expect no error, got %v", err)
+	}
+
+	response, ok := out.RawResponse.(*smithyhttp.Response)
+	if !ok {
+		t.Errorf("expect smithy response, got %T", out.RawResponse)
+	}
+
+	b := make([]byte, 100)
+	_, err = response.Body.Read(b)
+	if err != nil {
+		t.Log(err)
```

Comment:
Thanks for the update. Instead of logging the error, we should assert the type and expected error. 

```
readTimeoutError := &ReadTimeoutError{}
if !errors.Is(err, &readTimeoutError) {
	t.Fatalf("expect %T error, got %T", readTimeoutError, err)
}

```